frontend/src/components/BufferingDots.tsx

```
@@ -0,0 +1,44 @@
+    // Define size mapping
+    const sizeMap = {
+        small: 6,  // Small dot size
+        medium: 10,  // Medium dot size (default)
+        large: 14,  // Large dot size
+    };
+
+    const dotSize = sizeMap[size];  // Get the dot size based on the size prop
+
+    return (
+        <Box sx={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
+            <Box sx={dotStyle(1, theme.palette.primary.main, dotSize)}></Box>
```

Biome (1.9.4)

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)